### DISCUSSION TITLE
How to access ExecutionResult in IDocumentExecutionListener.AfterExecutionAsync?

### DISCUSSION THREAD

**Discussion by vincent-duret** [graphql-dotnet/graphql-dotnet#4074] [Q&A] [answered]
In some scenarii, I might want to modify the ExecutionResult before it is sent back to the client.
I was investigating to add a custom IDocumentExecutionListener to achieve this but found no way to access the ExecutionResult in the AfterExecutionAsync method. E.g. ValidationResult is passed as a parameter to AfterValidationAsync.

Is there a way to achieve this? Either using a document listener or any other strategy.

Thank you for your help

**Comment by Shane32** [comment, marked as answer]
```cs
services.AddGraphQL(b => b
    /* AddSchema and other calls */
    .ConfigureExecution(async (options, next) =>
    {
        // change the options
        var executionResult = await next(options);
        // do some stuff with the result
        return executionResult;
    })
);
```

**Comment by vincent-duret** [comment]
Thank you for your quick answer.
Would it make any sense to have the result passed to the AfterExecutionAsync method?

**Reply by Shane32** [reply]
The document listener takes place during execution phase. This code runs after execution finishes. It can provide greater control of the result when errors occur and so on, but less access to internal state. You can have as many document listeners configured as you wish and as many of the above snippets configured as you wish. They will all be called in order.
